pkg/operator/configobservation/satokencerts/observe_satokencerts.go

```diff
@@ -0,0 +1,79 @@
+package satokencerts
+
+import (
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+)
+
+// ObserveSATokenCerts checks which configmaps exist and bases the configuration for verifying sa tokens on them.
+// There are two possible sources: the installer and the kube-controller-manger-operator, but we wire to the target namespace
+// to avoid setting a config we cannot fulfill which would crash the kube-apiserver.
+func ObserveSATokenCerts(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
+	errs := []error{}
+	prevObservedConfig := map[string]interface{}{}
+
+	saTokenCertsPath := []string{"serviceAccountPublicKeyFiles"}
+	existingSATokenCerts, _, err := unstructured.NestedStringSlice(existingConfig, saTokenCertsPath...)
```

Comment:
the second return value says whether the key was found. It would make more sense to use that below in the if clause.

Reply:
> the second return value says whether the key was found. It would make more sense to use that below in the if clause.

I don't think so.  I think zero length values should be fully removed.